### PR TITLE
Outdoor Air Details Report - part 2

### LINKED ISSUE #8550: Outdoor Air Details Followup
Issue overview
--------------
Unfinished items from #8529. Most of these were addressed in #8596 unless noted otherwise below.

 - [x] Documentation - Output Details and Examples
 - [x] Populate Times for Outdoor Air Limiting Factors During Occupancy subtable
 - [x] Populate Average Outdoor Air For Each Outdoor Air Limiting Factors During Occupancy subtable
 - [ ] Possibly reorganize new variables - seems like lots of different places, not sure all the new variables are in appropriate places
 - [x] Look for efficiency improvements (8529 was about -0.06% performance)
 - [x] Unit tests - added in #8685 
 - [x] QA the numbers in the report

### Details
Some additional details for this issue (if relevant):
 - Version of EnergyPlus 9.5.0 I/O Freeze

### Checklist
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [x] Defect files added in EnergyplusDevSupport\DefectFiles
 - [x] Ticket added to Pivotal for defect (development team task)
 - [x] Pull request created (the pull request will have additional tasks related to reviewing changes that fix this defect)

### PR DESCRIPTION
**Pull request by mjwitte**
Pull request overview
---------------------
 - Fixes #8550
 - Complete the new Outdoor Air Details table report and clean up any issues found during testing.

### Pull Request Author
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [x] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [x] Label the PR with at least one of: Defect, Refactoring, NewFeature, Performance, and/or DoNoPublish
 - [ ] Pull requests that impact EnergyPlus code must also include unit tests to cover enhancement or defect repair
 - [ ] Author should provide a "walkthrough" of relevant code changes using a GitHub code review comment process
 - [ ] If any diffs are expected, author must demonstrate they are justified using plots and descriptions
 - [ ] If changes fix a defect, the fix should be demonstrated in plots and descriptions
 - [ ] If any defect files are updated to a more recent version, upload new versions here or on DevSupport
 - [ ] If IDD requires transition, transition source, rules, ExpandObjects, and IDFs must be updated, and add IDDChange label
 - [ ] If structural output changes, add to output rules file and add OutputChange label
 - [ ] If adding/removing any LaTeX docs or figures, update that document's CMakeLists file dependencies

### Reviewer
This will not be exhaustively relevant to every PR.
 - [ ] Perform a Code Review on GitHub
 - [ ] If branch is behind develop, merge develop and build locally to check for side effects of the merge
 - [ ] If defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [ ] If feature, test running new feature, try creative ways to break it
 - [ ] CI status: all green or justified
 - [ ] Check that performance is not impacted (CI Linux results include performance check)
 - [ ] Run Unit Test(s) locally
 - [ ] Check any new function arguments for performance impacts
 - [ ] Verify IDF naming conventions and styles, memos and notes and defaults
 - [ ] If new idf included, locally check the err file and other outputs

### CODE REVIEW

**Comment by mjwitte** [comment]
@Myoldmopar No unit tests yet, but otherwise this is ready for final review. If things are still open I can work on unit tests next week.

**Comment by Myoldmopar** [comment]
CI looks good, looking over the changes now.  I'm also building it locally to resolve the conflicts.

**Comment by Myoldmopar** [comment]
OK conflicts resolved, and built just fine.  Unit test suite passes fine, all integration tests pass fine.  Regressions look identical to the last set of CI results, lots of audit, rdd, and table diffs, nothing new.  Verified new values in 5ZoneAirCooled.  I think this is good to go.  Let me take a final peek at the code changes and this will likely merge.  Thanks @mjwitte

File: doc/output-details-and-examples/src/output-files/eplustbl-lt-ext-gt.tex [hunk 1]
```diff
@@ -1836,6 +1838,88 @@ \subsection{Outdoor Air Summary}\label{outdoor-air-summary}
...
+  \item[High Humidity] Controller:OutdoorAir High Humidity Control.
+  \item[Demand Controlled Ventilation] Controller:MechanicalVentilation requested more than the minimum outdoor air fraction.
+  \item[Night Ventilation] AvailabilityManager:NightVentilation is active.
+  \item[Demand Limiting] DemandManager:Ventilation has reduced the outdoor air flow rate.
+  \item[Energy Management System] An EnergyManagementSystem:Actuator has overridden the outdoor air flow rate.
+\end{description}
+
+\subsubsection{Average Outdoor Air for Limiting Factors During Occupancy}\label{average-outdoor-air-for-limiting-factors-during-occupancy}
+This subtable is the same as Outdoor Air Controller Limiting Factors by AirLoop except that it reports average flow rates in \si{\volumeFlowRate} at each limiting factor for occupied periods only.
+
```
Comment (Myoldmopar) [inline doc/output-details-and-examples/src/output-files/eplustbl-lt-ext-gt.tex:1921]: Good doc updates.

File: src/EnergyPlus/MixedAir.hh [hunk 2]
```diff
@@ -142,6 +142,18 @@ namespace MixedAir {
...
+    constexpr int limitFactorNone = 0;        // No limit other than fixed OA amount
+    constexpr int limitFactorLimits = 1;      // Limits and scheduled limits
+    constexpr int limitFactorEconomizer = 2;  // Economizer operation
+    constexpr int limitFactorExhaust = 3;     // Exhaust flow
+    constexpr int limitFactorMixedAir = 4;    // Mixed air flow rate
+    constexpr int limitFactorHighHum = 5;     // High humidity economizer control
+    constexpr int limitFactorDCV = 6;         // Demand-controlled ventilation
+    constexpr int limitFactorNightVent = 7;   // Night ventilation
+    constexpr int limitFactorDemandLimit = 8; // Demand-limiting
+    constexpr int limitFactorEMS = 9;         // EMS override
```
Comment (Myoldmopar) [inline src/EnergyPlus/MixedAir.hh:155]: OK

File: src/EnergyPlus/SystemReports.cc [hunk 3]
```diff
@@ -2037,28 +2042,28 @@ namespace EnergyPlus::SystemReports {
...
                                     state.dataAirSystemsData->PrimaryAirSystems(SysIndex).Name);
 
-                SetupOutputVariable(state,
+               SetupOutputVariable(state,
                                     "Air System Mechanical Ventilation Flow Rate",
                                     OutputProcessor::Unit::m3_s,
                                     state.dataSysRpts->SysMechVentFlow(SysIndex),
                                     "HVAC",
-                                    "Sum",
+                                    "Average",
```
Comment (Myoldmopar) [inline src/EnergyPlus/SystemReports.cc:2052]: Noticed this improvement in the rdd diffs, good.

File: src/EnergyPlus/SystemReports.hh [hunk 4]
```diff
@@ -127,29 +127,26 @@ namespace SystemReports {
...
+        Real64 SysTimeBelowVozDynTotalOcc = 0.0; // time [hrs] that mechanical+natural ventilation is < VozTarget - 1% during occupied
+        Real64 SysTimeAtVozDynTotalOcc = 0.0;    // time [hrs] that mechanical+natural ventilation is = VozTarget within 1% and > zero during occ
+        Real64 SysTimeAboveVozDynTotalOcc = 0.0; // time [hrs] that mechanical+natural ventilation is > VozTarget + 1% during occupied
+        Real64 SysTimeVentUnoccTotal = 0.0;      // time [hrs] that mechanical+natural ventilation is > zero during unoccupied
+        Real64 SysTimeOccupiedTotal = 0.0;       // time [hrs] that any zone is occupied
+
+        std::vector<Real64> SysTimeAtOALimit = {0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0};    // time [hrs] at limit [n]
+        std::vector<Real64> SysTimeAtOALimitOcc = {0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0}; // time [hrs] at limit [n] during occupied
+        std::vector<Real64> SysMechVentTotAtLimitOcc = {
+            0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0}; // air loop mech vent total vol OA at limit [n] {m3} during occupied
```
Comment (Myoldmopar) [inline src/EnergyPlus/SystemReports.hh:148]: :+1: :+1:

**Review by Myoldmopar — APPROVED** [review]

**Comment by Myoldmopar** [comment]
Everything looks great here.  Merging after I push the conflict resolved branch!

**Pull request merged**